### DISCUSSION TITLE
Takes for ever to launch a terminal

### DISCUSSION THREAD

**Discussion by jasonribble** [i3/i3#5720] [Q&A]
I only have this problem on one of my machines.
When I launch an app for the first time it takes a long time; sometimes up to 10 seconds to see the terminal or browser window

**Comment by orestisfl** [comment]
Launch apps from the terminal, is there any error message?

How do you start i3? Is it through xinit? If yes, share your xinitrc.

Are you using gnome terminal by any chance?

**Reply by jasonribble** [reply]
No error message. And i3 is started before I log in. I'm unsure if it's through xinit; how do I find out?

And yeah it's gnome terminal! Why do you think firefox is also taking time to launch?

**Reply by orestisfl** [reply]
What's the output of these commands:

```
cat /etc/locale.conf
```

```
localectl
```

```
cat ~/.xinitrc
```

```
cat ~/.xsession
```

Also, what operating system are you using and what kind of setup (DE, login manager)?

**Comment by jasonribble** [comment]
```
❯ cat /etc/locale.conf 
LANG=en_US.UTF-8
LC_ADDRESS=en_US.UTF-8
LC_IDENTIFICATION=en_US.UTF-8
LC_MEASUREMENT=en_US.UTF-8
LC_MONETARY=en_US.UTF-8
LC_NAME=en_US.UTF-8
LC_NUMERIC=en_US.UTF-8
LC_PAPER=en_US.UTF-8
LC_TELEPHONE=en_US.UTF-8
LC_TIME=en_US.UTF-8

~ 
home ❯ localectl
System Locale: LANG=en_US.UTF-8
    VC Keymap: (unset)         
   X11 Layout: us

~ 
home ❯ cat ~/.xsession
cat: /home/user/.xsession: No such file or directory

~ 
❯ cat ~/.xinitrc
exec i3

```

I'm using Manjaro Linux 23.0.2 currently. Linux kernel 5.13.19. My current environment is i3, no?

**Comment by jasonribble** [comment]
And I believe I'm using `gdm` for the login manager

**Comment by johnhomelander** [comment]
This happened with me as well. Firefox, thunar and some applications were
taking around 30 seconds to launch. This happened because I installed an
application which caused some error (I don't remember the error exactly,
I'll find out and share) with the applications. You could try deleting one
of the new applications you installed recently and see if that works. I'll
share the error I got when I ran firefox from terminal.

**Reply by jasonribble** [reply]
Yeah it looks like you're right. Check this out:

```
home ❯ firefox
[GFX1-]: vaapitest: ERROR
[GFX1-]: vaapitest: VA-API test failed: failed to initialise VAAPI connection.

ATTENTION: default value of option mesa_glthread overridden by environment.
ATTENTION: default value of option mesa_glthread overridden by environment.
```

I wonder what that's about

**Reply by johnhomelander** [reply]
Yes, I got this same error when running Firefox, its due to some misconfiguration (I guess). This happened with Thunar (File Explorer) and Thunderbird as well on my system . Started when I installed a screenshot tool - "shutter".

Found a solution for this on the web, it was a bit unreliable but worked. Check this - [https://support.mozilla.org/en-US/questions/1383497](link) - it involves adding a command to your i3 config to be executed when the system starts up.
This startup issue was permanently fixed when I uninstalled the application. I am yet to fully understand why this happens. If anyone has an explanation, that would be very helpful. I'll try to get to the bottom of this.

**Reply by jasonribble** [reply]
Very strange. I'll give this a go

**Comment by aleb** [comment]
See more workarounds at https://stackoverflow.com/a/77490432/804479
